apps/native-component-list/src/screens/BatteryScreen.tsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import { Alert, ScrollView, Text } from 'react-native';
+import * as Battery from 'expo-battery';
+
+import ListButton from '../components/ListButton';
+
+import HeadingText from '../components/HeadingText';
+import MonoText from '../components/MonoText';
+import { PowerState } from 'expo-battery/build/Battery.types';
+
+interface State {
+    batteryLevel?: number;
+    batteryState?: string;
+    powerState?: PowerState;
+    lowPowerMode?: string;
+}
+
+interface Subscription {
+    remove: () => void;
+}
+
+export default class BatteryScreen extends React.Component<{}, State> {
+    static navigationOptions = {
+        title: 'Battery',
+    };
+
+    readonly state: State = {};
+
+    _subscriptionPowerMode?: Subscription | null;
+    _subscriptionBatteryState?: Subscription | null;
+    _subscriptionBatteryLevel?: Subscription | null;
+
+
+    async componentDidMount() {
+        await this._pingBattery();
+        this._subscribePowerMode();
+        this._subscribeBatteryState();
+        this._subscribeBatteryLevel();
+    }
+
+    componentWillUnmount() {
+        this._unsubscribe();
+    }
+
+    _pingBattery = async () => {
+        let batteryLevel = await Battery.getBatteryLevelAsync();
```

Comment:
Use `const` instead of `let` whenever possible (you don't have to modify these variables, so it's okay to make them `const`)